src/ToggleIcon/VaToggleIcon.vue

```diff
@@ -58,24 +59,40 @@
     },
     data () {
       return {
-        actived: this.active,
+        isActive: this.active,
         style: {
           display: 'inline-block',
-          width: this.size,
-          height: this.size
+          width: this.$attrs.size,
+          height: this.$attrs.size,
+          position: 'relative'
         }
       }
     },
     methods: {
       toggle () {
-        this.actived = !this.actived
+        this.isActive = !this.isActive
       },
-      show () {
-        this.actived = true
+      onActivate () {
+        this.isActive = true
       },
-      hide () {
-        this.actived = false
+      onDeactivate () {
+        this.isActive = false
       }
     }
   }
 </script>
+
+<style>
+    .icon {
+        position: absolute;
+    }
+
+    .fade-enter-active, .fade-leave-active {
```

Comment:
a fade transition very similar to this is defined in /src/style/_animate.scss: https://github.com/nvms/vue-atlas/blob/master/src/style/_animate.scss#L101